### PR TITLE
Add Telegraf&NightWatch to DaemonSet

### PR DESCRIPTION
**Pull request by yangzb09**
For https://dashbase.atlassian.net/browse/DB-341
We add to Telegraf&NightWatch to DaemonSet

### CODE REVIEW

File: dashbase-k8s-solution/templates/filebeat.yaml [hunk 1]
```diff
@@ -88,6 +88,46 @@ data:
...
+        commands = ["night-watch watch-append -p '/var/lib/docker/containers/*/*.log'"]
+        timeout = "10s"
+        data_format = "influx"
+    # [[inputs.exec]]
+    #     commands = ["night-watch verify --api http://proxy:9200 --column 'host.name' --value $HOSTNAME --table kubernetes-logs -p '/var/lib/docker/containers/*/*.log' -f '2006-01-02T15:04:05.999999999Z07:00' -o 1800 -d 30"]
+    #     timeout = "1800s"
+    #     interval = "3600s"
+    #     data_format = "influx"
+    [[inputs.http]]
+        urls = ["http://localhost:5066/stats"]
```
Comment (sincejune) [inline dashbase-k8s-solution/templates/filebeat.yaml]: Will it work if you didn't open the port of Filebeat?

File: dashbase/templates/filebeat.yaml [hunk 2]
```diff
@@ -88,8 +88,50 @@ data:
...
+        commands = ["night-watch watch-append -p '/var/lib/docker/containers/*/*.log'"]
+        timeout = "10s"
+        data_format = "influx"
+    # [[inputs.exec]]
+    #     commands = ["night-watch verify --api http://proxy:9200 --column 'host.name' --value $HOSTNAME --table kubernetes-logs -p '/var/lib/docker/containers/*/*.log' -f '2006-01-02T15:04:05.999999999Z07:00' -o 1800 -d 30"]
+    #     timeout = "1800s"
+    #     interval = "3600s"
+    #     data_format = "influx"
+    [[inputs.http]]
+        urls = ["http://localhost:5066/stats"]
```
Comment (sincejune) [inline dashbase/templates/filebeat.yaml]: ditto

File: dashbase/templates/filebeat.yaml [hunk 3]
```diff
@@ -140,20 +182,47 @@ spec:
...
           readOnly: true
+      {{ if .Values.filebeat.podConfig }}
+{{ toYaml .Values.filebeat.podConfig | indent 6 }}
+      {{ end}}
+      - name: telegraf
+        image: dashbase/night-watch:v1.1.0
+        args: [
+          "--config", "/etc/telegraf.conf"
+        ]
+      {{ if .Values.telegraf.env }}
```
Comment (sincejune) [inline dashbase/templates/filebeat.yaml]: Can we use `environment` instead ?

File: dashbase/values.yaml [hunk 4]
```diff
@@ -30,6 +30,19 @@ filebeat:
         cpu: 100m
         memory: 100Mi
 
+telegraf:
+  enabled: false
```
Comment (sincejune) [inline dashbase/values.yaml]: Will `enabled` properly work?

File: dashbase/values.yaml [hunk 5]
```diff
@@ -30,6 +30,19 @@ filebeat:
...
+  global_tags:
+    source: filebeat-daemonset
```
Comment (sincejune) [inline dashbase/values.yaml]: We should require users to set tag `table` here.

File: dashbase-k8s-solution/templates/filebeat.yaml (hunk 1 above)
Comment (yangzb09) [reply dashbase-k8s-solution/templates/filebeat.yaml]: It can't, I let filebeat enable it.

File: dashbase/templates/filebeat.yaml (hunk 3 above)
Comment (yangzb09) [reply dashbase/templates/filebeat.yaml]: Yes I change it.

File: dashbase/values.yaml (hunk 4 above)
Comment (yangzb09) [reply dashbase/values.yaml]: We will enable it at deployment's values.yaml, It works.
https://github.com/dashbase/dashbase-deployment/pull/498/files

File: dashbase/values.yaml (hunk 5 above)
Comment (yangzb09) [reply dashbase/values.yaml]: I set "table: kubernetes-logs" as default value.

File: dashbase/templates/filebeat.yaml [hunk 6]
```diff
@@ -127,33 +173,54 @@ spec:
...
           runAsUser: 0
-      {{ if .Values.filebeat.containerConfig }}
-{{ toYaml .Values.filebeat.containerConfig | indent 8 }}
+      {{ if .Values.k8s_solution.filebeat.containerConfig }}
+{{ toYaml .Values.k8s_solution.filebeat.containerConfig | indent 8 }}
       {{ end}}
         volumeMounts:
         - name: config
           mountPath: /etc/filebeat.yml
-          readOnly: true
```
Comment (sincejune) [inline dashbase/templates/filebeat.yaml]: why did you remove this line?

File: dashbase/templates/filebeat.yaml [hunk 7]
```diff
@@ -86,10 +86,56 @@ data:
...
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: telegraf-config
+  labels:
+    k8s-app: telegraf
+data:
+  telegraf.conf: |-
+    [global_tags]
+        {{- range $key, $val := .Values.telegraf.global_tags }}
```
Comment (sincejune) [inline dashbase/templates/filebeat.yaml]: I think you should also modify here.

File: dashbase/values.yaml (hunk 5 above)
Comment (sincejune) [reply dashbase/values.yaml]: No, we should require users to set it manually.

File: dashbase/values.yaml (hunk 4 above)
Comment (sincejune) [reply dashbase/values.yaml]: What I want to say is that this `enabled` is useless because we have no corresponding code to handle it.

File: dashbase/templates/filebeat.yaml (hunk 6 above)
Comment (yangzb09) [reply dashbase/templates/filebeat.yaml]: I add it back.

Comment (yangzb09) [reply dashbase/templates/filebeat.yaml]: I add it back.

File: dashbase-k8s-solution/values.yaml [hunk 8]
```diff
@@ -1,14 +1,24 @@
...
-      memory: 200Mi
-    requests:
-      cpu: 100m
-      memory: 100Mi
+collection:
+  filebeat:
+    annotations:
+      dashbase.io/filebeat.parser.type: "simple"
+      dashbase.io/filebeat.parser.pattern: "iso"
+  containerConfig:
```
Comment (sincejune) [inline dashbase-k8s-solution/values.yaml]: `containerConfig` should under `filebeat` scope

File: dashbase/templates/filebeat.yaml [hunk 9]
```diff
@@ -86,10 +86,59 @@ data:
           fail_on_error: false
 
     output.elasticsearch:
-      hosts: ['{{ .Values.filebeat.elasticsearch_url }}']
+    {{ if .Values.collection.https  }}
+      hosts: ['https://table-kubernetes-logs.{{ .Values.collection.monitor_domain }}:443']
```
Comment (sincejune) [inline dashbase/templates/filebeat.yaml]: I'm wondering why it must start with `table-kubernetes-logs`?

**Comment by yangzb09** [comment]
Now, we set

File: dashbase-k8s-solution/values.yaml (hunk 8 above)
Comment (yangzb09) [reply dashbase-k8s-solution/values.yaml]: fixed it.

File: dashbase/templates/filebeat.yaml (hunk 9 above)
Comment (yangzb09) [reply dashbase/templates/filebeat.yaml]: Now, I set it back, we can define elasticsearch_url manually. It's more flexible.

File: dashbase/values.yaml [hunk 10]
```diff
@@ -16,19 +16,32 @@ ingress:
...
+      resources:
+        limits:
+          memory: 200Mi
+        requests:
+          cpu: 100m
+          memory: 100M
+  telegraf:
+    enabled: false
+    global_tags:
+      table: kubernetes-logs
```
Comment (sincejune) [inline dashbase/values.yaml]: I think we should require users' input on `table` here.
The others look great to me.

**Comment by sincejune** [comment]
One more thing, the `annotations` should be in the `collection` scope.
